include/experimental/__p1673_bits/blas1_dot.hpp

```diff
@@ -178,9 +192,9 @@ template<class ElementType1,
 auto dotc(
   std::experimental::mdspan<ElementType1, std::experimental::extents<ext1>, Layout1, Accessor1> v1,
   std::experimental::mdspan<ElementType2, std::experimental::extents<ext2>, Layout2, Accessor2> v2)
--> decltype(dot_detail::dot_return_type_deducer(v1, v2))
+-> decltype(dot_detail::dotc_return_type_deducer(v1, v2))
 {
-  using return_t = decltype(dot_detail::dot_return_type_deducer(v1, v2));
+  using return_t = decltype(dot_detail::dotc_return_type_deducer(v1, v2));
```

Comment:
Would you consider removing `dotc_return_type_deducer` and just having `dotc` return `auto`, as before?  Plain old `auto dotc` without an explicit return type should work just fine, because `dotc(x, y)` is just `dot(conjugated(x), y)`, and even C++14 knows how to deduce the function's return type from the `return` expression.

Reply:
@mhoemmen  
Just to clarify, before my changes, `dotc` looked like this:
```cpp
auto dotc(
  std::experimental::mdspan<ElementType1, std::experimental::extents<ext1>, Layout1, Accessor1> v1,
  std::experimental::mdspan<ElementType2, std::experimental::extents<ext2>, Layout2, Accessor2> v2)
-> decltype(dot_detail::dot_return_type_deducer(v1, v2))
```
so note that it was using `dot_return_type` , but WITHOUT saying `_deducer(conjugate(v1), v2)`. 
When I read the spec, I noticed this:
```
Effects: If in_vector_2_t::element_type is complex<R> for some R, let T be decltype(v1(0)*conj(v2(0))); 
else, let T be decltype(v1(0)*v2(0)). Then, the two-parameter overload is equivalent 
to dotc(v1, v2, T{});, and the three-parameter overload is equivalent to dotc(exec, v1, v2, T{});.
``` 
so that is why I added `dotc_return_type` with that syntax, to be consistent with what was written in the spec. 

So, assuming you still want to get rid of `dotc_return_type` do you want to leave:
```cpp
auto dotc(
  std::experimental::mdspan<ElementType1, std::experimental::extents<ext1>, Layout1, Accessor1> v1,
  std::experimental::mdspan<ElementType2, std::experimental::extents<ext2>, Layout2, Accessor2> v2)
-> decltype(dot_detail::dot_return_type_deducer(v1, v2))
{}
```
or should I just do:
```cpp
auto dotc(
  std::experimental::mdspan<ElementType1, std::experimental::extents<ext1>, Layout1, Accessor1> v1,
  std::experimental::mdspan<ElementType2, std::experimental::extents<ext2>, Layout2, Accessor2> v2)
  {}
```

Reply:
The latter two choices would be true to the proposal.  I would recommend the last choice: 
```cpp
auto dotc(
  std::experimental::mdspan<ElementType1, std::experimental::extents<ext1>, Layout1, Accessor1> v1,
  std::experimental::mdspan<ElementType2, std::experimental::extents<ext2>, Layout2, Accessor2> v2)
-> decltype(dot_detail::dot_return_type_deducer(conjugate(v1), v2))
```